### DISCUSSION TITLE
How to `update-right-status` on the event of `copied to clipboard`

### DISCUSSION THREAD

**Discussion by azzamsa** [wezterm/wezterm#4761] [Q&A] [answered]
### What Operating System(s) are you running on?

Linux Wayland

### Which Wayland compositor or X11 Window manager(s) are you using?

Gnome

### WezTerm version

20230712-072601-f4abf8fd

### Ask your question!

I use `update-right-status` to get some visual feedback.

But, is there any way to detect in the event of an item being copied to the clipboard?
Currently, I am using `toast`, but it is too noisy.

```lua
-- Copy text under selection then clear the selection.
function M.copy_and_clear_selection(window, pane)
    local has_selection = window:get_selection_text_for_pane(pane) ~= ""
    if has_selection then
        window:perform_action(act.CopyTo("ClipboardAndPrimarySelection"), pane)
        window:perform_action(act.ClearSelection, pane)
        window:toast_notification("Wezterm", "Text copied to clipboard.", nil, 2000)
    end
end
```

Is there something like:

```lua
wezterm.on("update-right-status", function(window, _)
    if window:item_copied_to_clipboard() then
        window:set_right_status("Copied to clipboard.")
    end
    window:set_right_status("")
end)
```

**Comment by wez** [comment, marked as answer]
What I'd suggest is looking at user vars.
https://wezfurlong.org/wezterm/recipes/passing-data.html

* Setting a user var will trigger the update-status event and you can pick up its contents from the `pane` object in the event callback
* In lua, you can use [pane:inject_output](https://wezfurlong.org/wezterm/config/lua/pane/inject_output.html) to emit the user var escape sequence into the pane to trigger the event.  This will only work for local panes; it is not compatible with multiplexer domain panes.
